products/platform/documentation-site/research/search-design-doc.md

```diff
@@ -1,15 +1,178 @@
+
+In general, this kind of system requires the following components:
+
+- Page with Input Textbox - We need a public page with a search input where users can input their search term(s).
+- Index/Database - We need a place to store the records that will be returned to the user.
+- Search API - We need a way to get data out of the index.
+- Crawler/Scraper - We need a crawler to scan our documentation sources and push data to our index.
+- Pipeline - We need a way to routinely run the crawler
+
+Each one of the above components could be a separate buy-or-build decision.
+
+By leveraging the following existing technologies, we should only need to add the search input to the landing page, configure the scraper, and configure when to run the scraper.
+
+- Algolia's Search API product will satisfy the index and API components.
+- Algolia's open source JavaScript snippet will display search results to the user.
+- Algolia's open source scraper will crawl and scrape our public documentation sources to push records into our index.
+
+#### Where/when to run the scraper
+
+For the MVP, we will run the crawler in a scheduled job once every 24 hours in Circle CI.
+
+#### Private repos
+
+---
+
+### Code Location
+
+The code will live in a new repo (tentatively titled something like `docs-landing-page`).
+
+### Testing Plan
+
+#### Environments
+
+For an MVP, I'm not certain it's necessary to have different environments for the search input and the scraper.
+
+##### Testing the scraper
+
+###### New records should appear in index after a crawl
+
+1. Search for term that returns zero results
+1. Add document that contains term
+1. Run crawler
+1. Search for recently added term
+1. Assert that term is in index
+
+###### Old records should disappear from index after a crawl
+
+1. Search for term that exists in index
+1. Remove document that contains term from index
+1. Run crawler
+1. Search for recently deleted term
+1. Assert that term isn't in index
+
+###### Records should reflect edits after a crawl
+
+1. Search for record that exists in index
+1. Edit document
+1. Run crawler
+1. Search for recently edited record
+1. Assert that record reflects recent edit
+
+##### Testing the search functionality
+
+Once we have records in our Algolia index and the DocSearch frontend library on a page, we should be able to test the search functionality the same way the end user would use it.
+
+### Logging
+
+Algolia includes analytics, such as ["Most popular searches and results most often returned by the search engine."](https://www.algolia.com/products/analytics/?tab=popular)
+
+### Debugging
+
+The Docker image for the scraper is available on [Docker Hub](https://hub.docker.com/r/algolia/docsearch-scraper). Once we have a config file, we can [run the scraper locally](https://community.algolia.com/docsearch/run-your-own.html#run-the-crawl-from-the-docker-image).
```

Comment:
This will populate the _actual_ database, right? Is there a way to run the scraper without writing to an actual database? Or maybe for this purpose, we have a docker image of the DB that it can write to instead of the live, hosted DB where we have to pay for records stored and operations performed?

Reply:
You are correct. The Docker image for the scraper will write to the actual DB. I haven't seen anything in their scraper docs to run the scraper without writing to the DB.